### PR TITLE
chore: refactor portfolio components to use redux

### PR DESCRIPTION
**Pull request by josheleonard**
Resolves https://github.com/brave/brave-browser/issues/22110

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
Expect no regressions within the portfolio view

### CODE REVIEW

**Review by Douglashdaniel — CHANGES_REQUESTED** [review]

**Comment by Douglashdaniel** [comment]
`SelectedAssets` price is persisting the last `hover` value.
It should change back to the `SelectedAssets` current price value on `mouseLeave`

https://user-images.githubusercontent.com/40611140/163029221-b2981299-1d1f-43d3-a1de-1429bcac612a.mov

**Comment by Douglashdaniel** [comment]
Screen Shot 2022-04-12 at 12 29 57 PM

`CustomReferenceDot` is not rendering for a `SelectedAsset`

File: components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx [hunk 1]
```diff
@@ -280,6 +223,81 @@ const PortfolioTransactionItem = (props: Props) => {
...
 
+  // custom hooks
+  const onClickViewOnBlockExplorer = useExplorer(transactionsNetwork)
+
+  // methods
+  const onShowTransactionPopup = () => setShowTransactionPopup(true)
+  const onClickCopyTransactionHash = React.useCallback(() => copyToClipboard(transaction.txHash), [transaction.txHash])
+  const onClickRetryTransaction = React.useCallback(() => onRetryTransaction(transaction), [transaction])
+  const onClickSpeedupTransaction = React.useCallback(() => onSpeedupTransaction(transaction), [transaction])
+  const onClickCancelTransaction = React.useCallback(() => onCancelTransaction(transaction), [transaction])
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx]: re-organized declarations

File: components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx [hunk 2]
```diff
@@ -280,6 +223,81 @@ const PortfolioTransactionItem = (props: Props) => {
...
+  }, [visibleTokens])
+
+  const onAssetClick = (symbol?: string) => () => {
+    if (!symbol) {
+      return
+    }
+
+    const asset = findToken(symbol)
+    if (asset) {
+      onSelectAsset(asset)
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx]: re-organized declarations

File: components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx [hunk 3]
```diff
@@ -280,6 +223,81 @@ const PortfolioTransactionItem = (props: Props) => {
...
+  const findToken = React.useCallback((symbol: string) => {
+    return visibleTokens.find((token) => token.symbol.toLowerCase() === symbol.toLowerCase())
+  }, [visibleTokens])
+
+  const onAssetClick = (symbol?: string) => () => {
+    if (!symbol) {
+      return
+    }
+
+    const asset = findToken(symbol)
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx]: re-organized declarations

File: components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx [hunk 4]
```diff
@@ -280,6 +223,81 @@ const PortfolioTransactionItem = (props: Props) => {
...
+    if (asset.contractAddress === '') {
+      history.push(`${WalletRoutes.Portfolio}/${asset.symbol}`)
+      return
+    }
+    history.push(`${WalletRoutes.Portfolio}/${asset.contractAddress}`)
+  }, [selectedTimeline])
+
+  const onSelectAccount = (account: WalletAccountType) => {
+    history.push(`${WalletRoutes.Accounts}/${account.address}`)
+  }
```
[the remark this answers is not in the document]
Comment (josheleonard) [reply components/brave_wallet_ui/components/desktop/portfolio-transaction-item/index.tsx]: re-organized declarations

**Review by Douglashdaniel — APPROVED** [review]
LGTM!

**Pull request merged**